src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOS_LandAssimGridComp.F90

```diff
@@ -2592,13 +2602,28 @@ subroutine get_mwrtm_param(internal,N_catl, rc)
     real, dimension(:), pointer :: BH
     real, dimension(:), pointer :: BV
     real, dimension(:), pointer :: LEWT
+    real, dimension(:), pointer :: DCATAU   ! needed?? declared as global var in module!?  reichle, 27Jun2021
     
     integer :: N_catl_tmp, n, mpierr, status
     logical :: mwp_nodata, all_nodata_l
-    
+
+    character(len=ESMF_MAXSTR)  :: TAUFile
+    type(ESMF_Time)             :: CURRENT_TIME
+ 
     if(allocated(mwRTM_param)) then
```

Comment:
@weiyuan-jiang, @gmao-qliu,
I'm grinding my way through the revisions of this PR and just stumbled across this "if" statement, which remains unchanged in Qing's branch.  If I understand the "if" statement here correctly, it looks like Qing's branch runs "get_mwrtm_param()" and thus reads DCATAU from the file ONLY ONCE per segment.  After "get_mwrtm_param()" has been allocated and filled, the DCATAU file is ignored.  Do I understand this correctly?

Reply:
The above understanding is consistent with the original use case for get_mwrtm_param() and also makes sense in that the variable mwRTM_param is allocated but never deallocated.
If this is all true, we need to fix the code and then re-run all the science tests from scratch, which would mean that this PR is no longer a candidate for the Version 6 L4_SM system. 
But maybe I'm missing something.  It's been a long day already.